File: Java/level 3/Nighttime.java
class NoOvertime {
	public int noOvertime(int no, int[] works) {
		int answer = 0;
		int cnt=0;
  	int max=0, index=0;
  	while(cnt<no){
  		for(int a=0; a<works.length; a++){
  			if(a==0){
      		max = works[a];
      		index = a;
    		}
      	if(max < works[a]){
      		max = works[a];
      		index = a;
    		}
  		}
    	works[index]-=1;
    	cnt++;
  	}
 	 	for(int b=0; b<works.length; b++)
    	answer += Math.pow(works[b],2);
		return answer;
	}
	public static void main(String[] args) {
		NoOvertime c = new NoOvertime();
		int []test = {4,3,3};
		System.out.println(c.noOvertime(4,test));
	}
}

File: Javascript/level 3/Nighttime.js
function noOvertime(no, works) {
	// 야근 지수를 최소화 하였을 때의 야근 지수는 몇일까요?
  var tem = 0;
	for(var i =0; i<no; i++){
    works[works.indexOf(Math.max.apply(null,works))] -= 1;
  }
	return works.reduce((result,x)=>result+Math.pow(x,2),0);
}